### ISSUE TITLE
Ruby 3?

### ISSUE THREAD

**Issue by MtnBiker** [postmodern/ruby-install#389]
Maybe I should be more patient but I'm getting this error from `ruby-install ruby`:

     Downloading https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.0.tar.bz2 into /Users/…/src ...
    --2020-12-26 10:52:57--  https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.0.tar.bz2
    Resolving cache.ruby-lang.org (cache.ruby-lang.org)... 151.101.41.178
    Connecting to cache.ruby-lang.org (cache.ruby-lang.org)|151.101.41.178|:443... connected.
    HTTP request sent, awaiting response... 404 Not Found

**Comment by aonemd** [comment]
Here to confirm the issue.

**Comment by larouxn** [comment]
This seems to be because `ruby-install` tries to download `tar.bz2`s but there isn't one available for Ruby `3.0.0`.
<img width="934" alt="Screen Shot 2020-12-27 at 17 21 30" src="https://user-images.githubusercontent.com/1557529/103166700-073a6c00-4868-11eb-9fcf-f37a88c9065e.png">
<img width="570" alt="Screen Shot 2020-12-27 at 17 21 43" src="https://user-images.githubusercontent.com/1557529/103166721-289b5800-4868-11eb-90ac-aa3422177b73.png">
Strangely, `tar.bz2`s exist for `3.0.0-preview1` and seemingly all prior full release Ruby versions such as `2.7.2`.
<img width="522" alt="Screen Shot 2020-12-27 at 17 25 18" src="https://user-images.githubusercontent.com/1557529/103166755-8465e100-4868-11eb-8bbd-abfcb52b6708.png">

**Comment by robzolkos** [comment]
Update to the latest `ruby-install` `0.8.1` and it will work fine.   On MacOS with homebrew `brew upgrade ruby-install`

**Comment by meineerde** [comment]
For reference: this was changed/fixed in a86f18bbf3bfe271e870f2b63df6248020273c1d and is released starting from version 0.8.0.

**Comment by MtnBiker** [comment]
@robzolkos Worked for me. I thought all the brews were updating. I wasn't paying attention between `update` and `upgrade`. update: only brew itself. upgrade: upgrade all the brews. English isn't precise enough for the difference.

**Comment by larouxn** [comment]
> Update to the latest `ruby-install` `0.8.1` and it will work fine. On MacOS with homebrew `brew upgrade ruby-install`

Confirmed, was able to install Ruby `3.0.0` after upgrading. Thanks! 🙌 

> \>>> Successfully installed ruby 3.0.0 into /Users/larouxn/.rubies/ruby-3.0.0

**Comment by kvz** [comment]
I upgraded to 0.8.1 but still got the warning:

```bash
ruby-install: 0.8.1
*** Unknown ruby version 3.0.0. Proceeding anyways ...
```

and then it does go on to compile as it states

**Comment by tejasbubane** [comment]
Folks from here could you share your MacOS and xcode versions over at https://github.com/postmodern/ruby-install/issues/390 please?

**Comment by marcandre** [comment]
False alarm, #390 is resolved.

**Comment by stillhart** [comment]
.tar.bz2 has been removed as target in current and future versions of ruby, see https://bugs.ruby-lang.org/issues/16483#note-7

It means the only way is to upgrade ruby-install.

**Comment by chrisseaton** [comment]
I still seem to get this error on 0.8.1:

```
% ruby-install --version
ruby-install: 0.8.1
% ruby-install ruby-3.0 
*** Unknown ruby version 3.0. Proceeding anyways ...
>>> Installing ruby 3.0 into /Users/chrisseaton/.rubies/ruby-3.0 ...
>>> Installing dependencies for ruby 3.0 ...
>>> Downloading https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.tar.xz into /Users/chrisseaton/src ...
--2021-01-09 14:43:24--  https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.tar.xz
Resolving cache.ruby-lang.org (cache.ruby-lang.org)... 199.232.53.178
Connecting to cache.ruby-lang.org (cache.ruby-lang.org)|199.232.53.178|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2021-01-09 14:43:24 ERROR 404: Not Found.

!!! Download of https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.tar.xz failed!
```

Trying https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.tar.xz manually gives me

```
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>pub/ruby/3.0/ruby-3.0.tar.xz</Key>
<RequestId>29013B321E235D46</RequestId>
<HostId>
VX88KMo5DhqcmxLpM0fMZ9GN8BouFkGNVloeuIwod4ennNtdiI/IRACeZ3wDSJ9RxO12l/c8W70=
</HostId>
</Error>
```

**Comment by NickLaMuro** [comment]
@chrisseaton If you look at the cache at the directory level:

https://cache.ruby-lang.org/pub/ruby/3.0/

You will see that you will need to add a patch version of `.0` to the end of your ruby version in your `ruby-install` call, and it should work just fine:

```diff
-% ruby-install ruby-3.0
+% ruby-install ruby-3.0.0
```

* * *

P.S.  Thanks to this thread pointing me to needing to upgrade, instead of me trying to hack together non-bz2 support myself 😆

**Comment by postmodern** [comment]
@chrisseaton @NickLaMuro normally `ruby-install ruby-3.0` should resolve to `3.0.X`. I suspect it's due to the versions list being outdated, so `3.0` is being interpreted as a literal version that's not yet known. `ruby-install` could possibly try to auto-update the ruby-versions files if the version cannot be resolved, then try again.

**Comment by postmodern** [comment]
This is related to #366 which was released in 0.8.0. Upgrade to ruby-install >= 0.8.0.
